### PR TITLE
[Algorithms] Add Clip-Cov and KL-Cov loss functions

### PR DESCRIPTION
**Pull request by SumanthRH**
# What does this PR do?

Adds Clip-Cov and KL-Cov loss functions based on https://arxiv.org/pdf/2505.22617 

TODO: 
- [ ] Add CPU test
- [ ] Cleanup

### CODE REVIEW

File: skyrl-train/skyrl_train/utils/ppo_utils.py [hunk 1]
```diff
@@ -607,6 +609,123 @@ def gspo_policy_loss(
...
+    ratio = torch.exp(negative_approx_kl)
+
+    pg_losses1 = -advantages * ratio
+    pg_losses_kl = -advantages * ratio + ppo_kl_coef * abs_kl
+    pg_losses = pg_losses1.clone()
+
+    all_valid = loss_mask > 0
+    all_valid_idx = torch.nonzero(all_valid.reshape(-1), as_tuple=True)[0]
+    all_valid_adv = advantages[all_valid].detach().reshape(-1).cpu()
+    all_valid_logp = log_probs[all_valid].detach().reshape(-1).cpu()
```
[the remark this answers is not in the document]
Comment (SumanthRH) [reply skyrl-train/skyrl_train/utils/ppo_utils.py:705]: Hmm this is same as reference, shouldn't have  an adverse effect

File: skyrl-train/examples/algorithms/clip_cov_kl_cov/run_clip_cov.sh [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+LOGGER="wandb"  # change to "console" to print to stdout
+
+# Configure Clip-Cov parameters
+POLICY_LOSS="clip_cov"
+
+uv run --isolated --extra vllm skyrl_train.train \
+  data.train_data="['$DATA_DIR/train.parquet']" \
+  data.val_data="['$DATA_DIR/validation.parquet']" \
+  trainer.algorithm.policy_loss_type="$POLICY_LOSS" \
+  trainer.algorithm.clip_cov.clip_ratio=0.0002 \
```
Comment (erictang000) [inline skyrl-train/examples/algorithms/clip_cov_kl_cov/run_clip_cov.sh]: tiny nit, but maybe move these flags with placeholder under the "# Configure Clip-Cov parameters" comment?

File: skyrl-train/examples/algorithms/clip_cov_kl_cov/run_kl_cov.sh [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+LOGGER="wandb"  # change to "console" to print to stdout
+
+# Configure KL-Cov parameters
+POLICY_LOSS="kl_cov"
+
+uv run --isolated --extra vllm skyrl_train.train \
+  data.train_data="['$DATA_DIR/train.parquet']" \
+  data.val_data="['$DATA_DIR/validation.parquet']" \
+  trainer.algorithm.policy_loss_type="$POLICY_LOSS" \
+  trainer.algorithm.kl_cov.kl_cov_frac=0.2 \
```
Comment (erictang000) [inline skyrl-train/examples/algorithms/clip_cov_kl_cov/run_kl_cov.sh]: same nit here

File: skyrl-train/docs/configuration/config.rst [hunk 4]
```diff
@@ -315,6 +315,17 @@ Algorithm Configuration
       # dual clip parameters
       clip_ratio_c: 3.0
 
+      # clip-cov parameters (only used when policy_loss_type: "clip_cov")
```
Comment (erictang000) [inline skyrl-train/docs/configuration/config.rst:318]: nit: can you update the example config in the docs (above this) to include clip_cov and kl_cov also

like this guy:
policy_loss_type: "regular" # "regular", "dual_clip", "gspo", or customizable with PolicyLossRegistry

**Comment by erictang000** [comment]
can you also make sure training runs look normal for a few steps with these guys

**Comment by SumanthRH** [comment]
Testing for just 10 steps, convergence differs a bit but expected. Impl matches reference we should be good: 

Screenshot 2025-09-08 at 2 02 18 PM
